### PR TITLE
Prevent events map from being re-rendered every 5s

### LINKED ISSUE #446: Event map redraws when data changes
Full map is redrawn when a marker moves on the event map causing the map to zoom to home coordinates then back to bounds of markers.

to reproduce: 

have an event open with a moving tracked user.

### LINKED ISSUE #466: Job Acknowledgment - Event Response
When responding to immediate / rescue jobs an acknowledge job should be implemented for Unit Controllers to accept jobs

### PR DESCRIPTION
**Pull request by OSPFNeighbour**
Rewrote previous componentWillReceiveProps function into getDerivedStateFromProps.

Minor timer name changes and white space changes.

I was incorrectly returning from the function which was causing the map to get redrawn every 5 seconds.

Fixes #446

### CODE REVIEW

File: client/src/screens/events/Detail.js [hunk 1]
```diff
@@ -71,6 +71,24 @@ class Detail extends Component {
     return mapMarkers;
   }
 
+  static getDerivedStateFromProps(newProps) {
```
Comment (sdunster) [inline client/src/screens/events/Detail.js]: i'm not going to block this PR if you think it improves it but...
Why do we implement this at all? This should be able to be done in render with low CPU penalty. We should avoid storing state in the component unless absolutely necessary or we actually have a bad perf problem. Leave it for now - I'll have a go at sorting it out later once more critical stuff is done.

**Comment by sdunster** [comment]
I fixed your title

File: client/src/screens/events/Detail.js (hunk 1 above)
Comment (OSPFNeighbour) [reply client/src/screens/events/Detail.js]: If we hold it in state then the individual map makers can be notified of a state change if needed.

If we hold it in 'this' or generate it at render time we are pushing new markers every time.

Well thats my understanding and that was done in a PR long ago, unrelated to this specific change directly.

Comment (sdunster) [reply client/src/screens/events/Detail.js]: I don't think that's how it works. As long as the markers you're rendering are exactly the same (attributes/properties) there should be no need to re-init the map.

**Pull request merged**